nixos/modules/services/misc/octoprint.nix

```diff
@@ -122,6 +122,9 @@ in
         ExecStart = "${pluginsEnv}/bin/octoprint serve -b ${cfg.stateDir}";
         User = cfg.user;
         Group = cfg.group;
+        SupplementaryGroups = [
+          "dialout"
```

Comment:
Did you consider adding an option like the one gammu has that defaults to "dialout" to avoid hardcoding? e.g. `services.octoprint.deviceGroup`

Reply:
No, because `dialout` is really the default for these kinds of devices. Not hardcoding would be a good idea, if `dialout` wasn't the common case here, but is is, so I'm somewhat against introducing more options.

In fact I'm not even sure we should provide `services.octoprint.user` and `services.octoprint.group`.

Reply:
My only concern was to allow users to change that dialout group, but I agree that's an advanced operation and that probably they will know how to override a module at that point.
As for the User and Group I noticed that many services are being converted to DynamicUser=true and that most of them have no persistent state so I assume it's unrelated to the device access and it concerns who can read/write the state directory more